File: 189-RotateArray/189-RotateArray.cpp
// Last updated: 07/04/2026, 11:40:29
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        k = k%n;
        reverse(nums.begin(),nums.end());
        reverse(nums.begin(),nums.begin()+k);
        reverse(nums.begin()+k,nums.end());
    }
};

File: 189-RotateArray/189-RotateArray.java
// Last updated: 09/04/2026, 10:38:01
1class Solution {
2    public void reverse(int [] nums, int i , int j){
3        while(i <j){
4            int temp = nums[i];
5            nums[i++] = nums[j];
6            nums[j--] = temp;
7
8        }
9    }
10    public void rotate(int[] nums, int k) {
11            int n = nums.length;
12            k = k%n;
13            reverse(nums,0,n-1);
14            reverse(nums,0,k-1);
15            reverse(nums,k,n-1);
16    }
17}